File: src/main/java/cz/cvut/fel/plichjan/distmesh/matlab/Bar.java
package cz.cvut.fel.plichjan.distmesh.matlab;

/**
 * Edge from point "a" to point "b".
 */
public class Bar implements Comparable<Bar> {
    protected int a;
    protected int b;

    public Bar(int a, int b) {
        this.a = Math.min(a, b);
        this.b = Math.max(a, b);
    }

    public int getA() {
        return a;
    }

    public void setA(int a) {
        this.a = a;
    }

    public int getB() {
        return b;
    }

    public void setB(int b) {
        this.b = b;
    }

    @Override
    public int compareTo(Bar o) {
        return a < o.getA() ? -1 : a > o.getA() ? 1 : b < o.getB() ? -1 : b > o.getB() ? 1 : 0;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Bar)) {
            return false;
        }

        Bar bar = (Bar) o;

        if (a != bar.a) {
            return false;
        }
        if (b != bar.b) {
            return false;
        }

        return true;
    }

    @Override
    public int hashCode() {
        int result = a;
        result = 31 * result + b;
        return result;
    }

    @Override
    public String toString() {
        return "Bar{" +
                "a=" + a +
                ", b=" + b +
                '}';
    }
}

File: src/main/kotlin/cz/cvut/fel/plichjan/distmesh/matlab/Bar.kt
package cz.cvut.fel.plichjan.distmesh.matlab

import kotlin.math.max
import kotlin.math.min

/**
 * Edge from point "a" to point "b".
 */
open class Bar(a: Int, b: Int) : Comparable<Bar> {
    var a: Int = min(a, b)
    var b: Int = max(a, b)

    override fun compareTo(other: Bar): Int {
        return when {
            a < other.a -> -1
            a > other.a -> 1
            b < other.b -> -1
            b > other.b -> 1
            else -> 0
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Bar) return false

        if (a != other.a) return false
        if (b != other.b) return false

        return true
    }

    override fun hashCode(): Int {
        var result = a
        result = 31 * result + b
        return result
    }

    override fun toString(): String {
        return "Bar(a=$a, b=$b)"
    }
}
